pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -377,12 +393,19 @@ func (bfs *Service) fetchOIDsBySearch() error {
 	startIndex := bfs.heightFunc()
 	batchSize := uint32(neofs.DefaultSearchBatchSize)
 
+	if bfs.stopAt > 0 && startIndex >= bfs.stopAt {
+		return nil
+	}
```

Comment:
This condition should be moved to `default` case of `for` loop. Then you'll have a single `stopAt`-related check per the whole `fetchOIDsBySearch`.